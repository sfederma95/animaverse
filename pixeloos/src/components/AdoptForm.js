import React, {useState, useContext} from 'react';
import {useHistory} from 'react-router-dom';
import Alert from '../Alert'
import AnimalsApi from '../api'
import UserContext from '../users/UserContext'

function AdoptForm({src, goBack}){
    const {currentUser} = useContext(UserContext)
    const history = useHistory();
    const [formData,setFormData]=useState({
        pet_name: "",
        pet_img: src
    })
    const [formErrors,setFormErrors] = useState([])
    async function handleSubmit(e){
        try{
            e.preventDefault();
            await AnimalsApi.newPet(formData,currentUser.usr_id)
            history.push(`/users/${currentUser.usr_id}`)
        } catch(e){
            let errors = [e]
            setFormErrors(errors)
        }
    }
    function handleChange(e){
        const {name,value} = e.target;
        setFormData(l=>({...l,[name]:value}))
    }
    return(
        <div id='whole-form'>
            <img className='chosen-pet' alt='adopt-pet' src={src}/>
            <form id='form-el' onSubmit={handleSubmit}>
                <div className='inputs'>
                    <label className='labels' htmlFor='pet_name'>Pet Name:</label>
                    <input className='input-el good-input' id='pet_name' name='pet_name' value={formData.pet_name} onChange={handleChange} type='text' required/>
                </div>
                {formErrors.length ? <Alert messages={formErrors}/> : null}
                <button className='adopt-btns'>Create</button>
                <button className='adopt-btns' onClick={goBack}>Back</button>
            </form>
        </div>
    )
}

export default AdoptForm;