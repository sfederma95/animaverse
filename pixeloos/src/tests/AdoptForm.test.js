import React from 'react';
import {render} from '@testing-library/react';
import {MemoryRouter} from 'react-router';
import AdoptForm from '../components/AdoptForm';
import {UserProvider} from './testSetup';

it ('renders choose pet component without crashing', function(){
    render(
        <MemoryRouter>
            <UserProvider>
                <AdoptForm src='rat.png'/>
            </UserProvider>
        </MemoryRouter>
    )
})

it ('matches snapshot', function(){
    const {asFragment} = render (
        <MemoryRouter>
            <UserProvider>
                <AdoptForm src='rat.png'/>
            </UserProvider>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot();
})