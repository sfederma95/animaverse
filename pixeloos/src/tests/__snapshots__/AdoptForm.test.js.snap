// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    id="whole-form"
  >
    <img
      alt="adopt-pet"
      class="chosen-pet"
      src="rat.png"
    />
    <form
      id="form-el"
    >
      <div
        class="inputs"
      >
        <label
          class="labels"
          for="pet_name"
        >
          Pet Name:
        </label>
        <input
          class="input-el good-input"
          id="pet_name"
          name="pet_name"
          required=""
          type="text"
          value=""
        />
      </div>
      <button
        class="adopt-btns"
      >
        Create
      </button>
      <button
        class="adopt-btns"
      >
        Back
      </button>
    </form>
  </div>
</DocumentFragment>
`;
